src/components/ProjectsTable/ProjectsTable.tsx

```diff
@@ -1,41 +1,48 @@
 import { ProjectsTableRow } from "./ProjectsTableRow";
 import { ProjectsTableHeadRow } from "./ProjectsTableHeadRow";
 import { useProjectsContext } from "@/contexts/projectsContext";
-import { useEffect, useState } from "react";
+import loadingImg from "@/assets/loading.gif";
 
 type TableProps = {
   filter: string;
 };
 
 export function ProjectsTable(props: TableProps) {
-  const { projects } = useProjectsContext();
-  const [filtered, setFiltered] = useState(projects);
-
-  useEffect(() => {
-     
-    const approved = projects.filter((project) => project.status === "approved");
-
-    const typed = approved.filter((project) => project.metadata.sunnyAwards.projectType === props.filter);
-    
-    setFiltered(typed);
-
+  const { projects, isPending, isError } = useProjectsContext();
+
+  if (isPending) {
+    return (
+      <div className="text-center">
+        <img
+          src={loadingImg}
+          alt="Loading..."
+          style={{ height: 20, margin: "0 auto" }}
+        />
+      </div>
+    );
   }
-    , [projects]);
-
 
+  if (isError) {
+    return <div className="text-center">Something went wrong</div>;
+  }
 
   return (
     <div className="relative pt-12 overflow-x-auto">
       <table className="w-full text-xl text-black border-collapse">
         <ProjectsTableHeadRow />
         <tbody>
-          {filtered.map((project, index) => (
-            <ProjectsTableRow
-              key={project.id}
-              index={index}
-              project={project}
-            />
-          ))}
+          {projects
+            .filter(
+              project =>
+                project.metadata.sunnyAwards.projectType === props.filter
+            )
```

Comment:
If we start to see rendering performance with many projects we can wrap this in a useMemo